apps/src/p5lab/layoutUtils.js

```diff
@@ -0,0 +1,177 @@
+ * https://github.com/code-dot-org/dance-party/blob/6672bdb0cffad1cbfda6e7396155f542b6cdcffe/src/p5.dance.js#L637-L911
+ */
+export function layoutSpriteGroup(group, layout, p5) {
+  const count = group.length;
+  if (!count) {
+    return;
+  }
+
+  const minX = 20;
+  const maxX = 400 - minX;
+  const minY = 35;
+  const maxY = 400 - 40;
+  const radiansToDegrees = 180 / Math.PI;
+  const maxCircleRadius = 165;
+
+  if (layout === 'top') {
+    createRow(group, 100);
+  } else if (layout === 'row') {
+    createRow(group, 200);
+  } else if (layout === 'bottom') {
+    createRow(group, 300);
+  } else if (layout === 'border') {
+    // First fill the four corners.
+    // Then split remainder into 4 groups. Distribute group one along the top,
+    // group 2 along the right, etc.
+    if (count > 0) {
+      group[0].x = minX;
+      group[0].y = minY;
+      group[0].rotation = 0;
+    }
+    if (count > 1) {
+      group[1].x = maxX;
+      group[1].y = minY;
+      group[1].rotation = 0;
+    }
+    if (count > 2) {
+      group[2].x = maxX;
+      group[2].y = maxY;
+      group[2].rotation = 0;
+    }
+    if (count > 3) {
+      group[3].x = minX;
+      group[3].y = maxY;
+      group[3].rotation = 0;
+    }
+    if (count > 4) {
+      const topCount = Math.ceil((count - 4 - 0) / 4);
+      const rightCount = Math.ceil((count - 4 - 1) / 4);
+      const bottomCount = Math.ceil((count - 4 - 2) / 4);
+      const leftCount = Math.ceil((count - 4 - 3) / 4);
+
+      for (let i = 0; i < topCount; i++) {
+        const sprite = group[4 + i];
+        // We want to include the corners in our total count so that the first
+        // inner sprite is > 0 and the last inner sprite is < 1 when we lerp.
+        sprite.x = p5.lerp(minX, maxX, (i + 1) / (topCount + 1));
+        sprite.y = minY;
+        sprite.rotation = 0;
+      }
+
+      for (let i = 0; i < rightCount; i++) {
+        const sprite = group[4 + topCount + i];
+        sprite.x = maxX;
+        sprite.y = p5.lerp(minY, maxY, (i + 1) / (rightCount + 1));
+        sprite.rotation = 0;
+      }
+
+      for (let i = 0; i < bottomCount; i++) {
+        const sprite = group[4 + topCount + rightCount + i];
+        sprite.x = p5.lerp(minX, maxX, (i + 1) / (bottomCount + 1));
+        sprite.y = maxY;
+        sprite.rotation = 0;
+      }
+
+      for (let i = 0; i < leftCount; i++) {
+        const sprite = group[4 + topCount + rightCount + bottomCount + i];
+        sprite.x = minX;
+        sprite.y = p5.lerp(minY, maxY, (i + 1) / (leftCount + 1));
+        sprite.rotation = 0;
+      }
+    }
+  } else if (layout === 'circle') {
+    // Adjust radius of circle and size of the sprite according to number of
+    // sprites in our group.
+    const pct = p5.constrain(count / 10, 0, 1);
+    const radius = p5.lerp(50, maxCircleRadius, pct);
+    const scale = p5.lerp(0.8, 0.3, pct * pct);
```

Comment:
Just noting that it looks like the "circle" format has its own logic for scale that overrides the `30` constant set at the top of the dance version of `layoutSprite`.